File: cpp/test/greet_test.cpp
#include "polyglot_app_template/greet.hpp"

#include <gtest/gtest.h>

TEST(Greet, ReturnsExpectedMessage) {
    EXPECT_EQ(polyglot_app_template::greet("Polyglot-App-Template"), "Hello, Polyglot-App-Template!");
}

TEST(Greet, HandlesDefaultCase) {
    EXPECT_EQ(polyglot_app_template::greet("world"), "Hello, world!");
}

File: go/internal/greet/greet_test.go
package greet

import "testing"

func TestGreet(t *testing.T) {
	cases := []struct {
		name string
		want string
	}{
		{"Polyglot-App-Template", "Hello, Polyglot-App-Template!"},
		{"world", "Hello, world!"},
	}

	for _, tc := range cases {
		if got := Greet(tc.name); got != tc.want {
			t.Errorf("Greet(%q) = %q, want %q", tc.name, got, tc.want)
		}
	}
}

func BenchmarkGreet(b *testing.B) {
	for i := 0; i < b.N; i++ {
		Greet("world")
	}
}
